src/infra/ssl/sslCmdRepo.go

```diff
@@ -2,135 +2,163 @@ package sslInfra
-  certFile   ` + sslCertFilePath + `
-  certChain  ` + sslCertChain +
-		vhsslChainedConfig + `
-}` + vhsslConfigBreakline
-
-	return vhsslConfig
+type SslCmdRepo struct {
+	sslQueryRepo SslQueryRepo
 }
 
-func (repo SslCmdRepo) Add(addSslPair dto.AddSslPair) error {
-	sslQueryRepo := SslQueryRepo{}
-
-	vhostConfigFilePath, err := sslQueryRepo.GetVhostConfigFilePath(addSslPair.Hostname)
-	if err != nil {
-		return err
+func NewSslCmdRepo() SslCmdRepo {
+	return SslCmdRepo{
+		sslQueryRepo: SslQueryRepo{},
 	}
+}
 
-	sslBaseDirPath := "/speedia/pki/" + addSslPair.Hostname.String()
-	sslKeyFilePath := sslBaseDirPath + "/ssl.key"
-	sslCertFilePath := sslBaseDirPath + "/ssl.crt"
-
-	err = infraHelper.MakeDir(sslBaseDirPath)
+func (repo SslCmdRepo) forceSymlink(
+	pkiSourcePath string,
+	pkiTargetPath string,
+) error {
+	err := os.Remove(pkiTargetPath)
 	if err != nil {
-		return err
+		return errors.New("FailedToDeletePkiFile: " + err.Error())
 	}
 
-	err = infraHelper.UpdateFile(sslCertFilePath, addSslPair.Certificate.String(), true)
+	err = os.Symlink(pkiSourcePath, pkiTargetPath)
 	if err != nil {
-		return err
+		return errors.New("AddPkiSymlinkError: " + err.Error())
 	}
 
-	err = infraHelper.UpdateFile(sslKeyFilePath, addSslPair.Key.String(), true)
-	if err != nil {
-		return err
-	}
+	return nil
+}
 
-	sslPairCertificate := addSslPair.Certificate
-	sslCertificates, err := sslQueryRepo.SslCertificatesFactory(
-		sslPairCertificate.Certificate,
+func (repo SslCmdRepo) SwapToSelfSignedCert(vhost valueObject.Fqdn) error {
+	selfSignedSslKeyPath := "/app/conf/pki/" + vhost.String() + ".key"
+	selfSignedSslCertPath := "/app/conf/pki/" + vhost.String() + ".crt"
+
+	_, err := infraHelper.RunCmd(
+		"openssl",
+		"req",
+		"-x509",
+		"-nodes",
+		"-days",
+		"365",
+		"-newkey",
+		"rsa:2048",
+		"-keyout",
+		selfSignedSslKeyPath,
+		"-out",
+		selfSignedSslCertPath,
+		"-subj",
+		"/C=US/ST=California/L=LosAngeles/O=Acme/CN="+vhost.String(),
 	)
 	if err != nil {
-		return err
+		return errors.New("SwapToSelfSignedCertFailed: " + err.Error())
 	}
 
-	newSsl, err := sslQueryRepo.SslPairFactory(
-		addSslPair.Hostname,
-		addSslPair.Key,
-		sslCertificates,
-	)
-	if err != nil {
-		return err
+	return nil
+}
+
+func (repo SslCmdRepo) Add(addSslPair dto.AddSslPair) error {
+	if len(addSslPair.VirtualHosts) == 0 {
+		return errors.New("NoVirtualHostsProvidedToAddSslPair")
 	}
 
-	isChainedCert := true
-	if len(newSsl.ChainCertificates) == 1 {
-		isChainedCert = false
+	firstVhostStr := addSslPair.VirtualHosts[0].String()
+	for _, vhost := range addSslPair.VirtualHosts {
+		vhostStr := vhost.String()
+		vhostCertFilePath := "/app/conf/pki/" + vhostStr + ".crt"
+		vhostCertKeyFilePath := "/app/conf/pki/" + vhostStr + ".key"
+
+		shouldBeSymlink := vhostStr != firstVhostStr
+		if shouldBeSymlink {
+			firstVhostCertFilePath := "/app/conf/pki/" + firstVhostStr + ".crt"
+			firstVhostCertKeyFilePath := "/app/conf/pki/" + firstVhostStr + ".key"
```

Comment:
Você pode declarar essas duas variáveis antes do laço for pois provavelmente ela será usada diversas vezes durante o laço.

Reply:
Na verdade a única forma correta era declarar fora mesmo. Não há motivo de eu ficar "redeclarando" isso toda hora dentro de um laço.

Ajustado!